java/src/org/openqa/selenium/json/Json.java

```diff
@@ -27,20 +27,117 @@
 import java.util.List;
 import java.util.Map;
 
+/**
+ * The <b>Json</b> class is the entrypoint for the JSON processing features of the Selenium API.
+ * <p>
+ * These features include:
+ * <ul>
+ *   <li>Built-in serialization of primitives and collections.</li>
+ *   <li>Facilities to serialize custom data types:
+ *     <ul>
+ *       <li>Classes that declare getter methods that adhere to the
+ *         <a href="https://docs.oracle.com/javase/tutorial/javabeans/writing/index.html">JavaBean</a>
+ *         specification.</li>
+ *       <li>Classes that declare a {@code toJson()} method that returns primitives and collections.</li>
+ *     </ul>
+ *   </li>
+ *   <li>Build-in deserialization of primitives and collections.</li>
+ *   <li>Facilities to deserialize custom data types:
+ *     <ul>
+ *       <li>Classes that declare setter methods that adhere to the {@code JavaBean} specification.</li>
+ *       <li>Classes that declare a {@code fromJson(JsonInput)} method.</li>
```

Comment:
Can also use a `fromJson(primitive)` method if needed, which makes it easier to have "[tiny types](https://darrenhobbs.com/2007/04/11/tiny-types/)"

Reply:
I added this, and I also noted the potential for producing immutable objects. I wasn't sure about how to document "tiny types", though.